projects/swimlane/ngx-ui/src/lib/components/select/select.component.ts

```diff
@@ -123,7 +123,7 @@ export class SelectComponent implements ControlValueAccessor, OnDestroy {
 
   @HostBinding('class.invalid')
   get invalid() {
-    if (this.required && (!this.value || this.value.length < 1)) return true;
+    if (this.required && (!this.value || this.value.length < 1 || (this.value.length === 1 && !this.value[0]))) return true;
```

Comment:
`!this.value[0]`

Would we have an issue here if the value is `0`?

Reply:
Yup, same issue in the previous statement `!this.value`. I'll compare with `undefined`.